app/main.py

```diff
@@ -1,8 +1,47 @@
+from typing import List
+
+
 class Car:
-    # write your code here
-    pass
+    def __init__(self, comfort_class: int,
+                 clean_mark: int, brand: str) -> None:
+        self.comfort_class = comfort_class
+        self.clean_mark = clean_mark
+        self.brand = brand
 
 
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(self, distance_from_city_center: float, clean_power: int,
+                 average_rating: float, count_of_ratings: int) -> None:
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = average_rating
+        self.count_of_ratings = count_of_ratings
+
+    def serve_cars(self, cars: List[Car]) -> float:
+        total_income = 0.0
+        for car in cars:
+            if car.clean_mark < self.clean_power:
+                total_income += self.calculate_washing_price(car)
+                self.wash_single_car(car)
+
+        return round(total_income, 1)
+
+    def calculate_washing_price(self, car: Car) -> float:
+        cost = (car.comfort_class * (self.clean_power - car.clean_mark)
+                * self.average_rating / self.distance_from_city_center)
+        return round(cost, 1)
+
+    def wash_single_car(self, car: Car) -> None:
+        if self.clean_power > car.clean_mark:
+            car.clean_mark = self.clean_power
+
+    def rate_service(self, rating: float) -> None:
+        total_ratings = self.average_rating * self.count_of_ratings
+        self.count_of_ratings += 1
+        self.average_rating = (total_ratings + rating) / self.count_of_ratings
+        self.average_rating = round(self.average_rating, 1)
```

Comment:
that's right, but so complicated, consider either break it down in smaller functions, which will show, what done there, or rewright it, so flow will be right.
Also I don't think it's necessary to make 1 more row to only round the answer, here, consider:
```suggestion
    def rate_service(self, rating: float) -> None:
        total_ratings = self.average_rating * self.count_of_ratings
        self.count_of_ratings += 1
        self.average_rating = round((total_ratings + rating) / self.count_of_ratings, 1)
```
or something similar.